Classes/common/touchdb/TD_DatabaseManager.m

```diff
@@ -117,39 +117,60 @@ - (NSString*)pathForName:(NSString*)name
     return [_dir stringByAppendingPathComponent:[name stringByAppendingPathExtension:kDBExtension]];
 }
 
-- (TD_Database*)databaseNamed:(NSString*)name create:(BOOL)create
+- (TD_Database*)databaseNamed:(NSString*)name
+            withEncryptionKey:(id<CDTEncryptionKey>)encryptionKey
+                       create:(BOOL)create
 {
-    if (_options.readOnly) create = NO;
+    if (_options.readOnly) {
+        create = NO;
+    }
+
     TD_Database* db = _databases[name];
     if (!db) {
         NSString* path = [self pathForName:name];
-        if (!path) return nil;
-        db = [[TD_Database alloc] initWithPath:path];
+        if (!path) {
+            return nil;
+        }
+
+        db = [[TD_Database alloc] initWithPath:path encryptionKey:encryptionKey];
         db.readOnly = _options.readOnly;
         if (!create && !db.exists) {
             return nil;
         }
+
         db.name = name;
         _databases[name] = db;
     }
+
     return db;
 }
 
-- (TD_Database*)databaseNamed:(NSString*)name { return [self databaseNamed:name create:YES]; }
+- (TD_Database*)databaseNamed:(NSString*)name withEncryptionKey:(id<CDTEncryptionKey>)encryptionKey
+{
+    return [self databaseNamed:name withEncryptionKey:encryptionKey create:YES];
+}
 
 - (TD_Database*)existingDatabaseNamed:(NSString*)name
+                    withEncryptionKey:(id<CDTEncryptionKey>)encryptionKey
 {
-    TD_Database* db = [self databaseNamed:name create:NO];
-    if (db && ![db open]) db = nil;
+    TD_Database* db = [self databaseNamed:name withEncryptionKey:encryptionKey create:NO];
+    if (db && ![db open]) {
+        db = nil;
+    }
+
     return db;
 }
 
-- (BOOL)deleteDatabaseNamed:(NSString*)name
+- (BOOL)deleteDatabaseNamed:(NSString*)name withEncryptionKey:(id<CDTEncryptionKey>)encryptionKey
 {
-    TD_Database* db = [self databaseNamed:name];
-    if (!db) return NO;
+    TD_Database* db = [self databaseNamed:name withEncryptionKey:encryptionKey];
```

Comment:
This seems sort of wrong that you need to pass in the encryption key to delete the database (because the `databaseNamed` lookup requires it). Maybe we need a more lightweight lookup just to check the database exists without having to pass the key in.